src/pip/_internal/cli/autocompletion.py

```
@@ -7,125 +7,154 @@
 import sys
 from collections.abc import Iterable
 from itertools import chain
-from typing import Any
+from typing import Any, Optional, List, Tuple
```

Ruff (0.11.9)

`typing.List` is deprecated, use `list` instead

(UP035)